### PR TITLE
Implement the Advanced tab of new room settings

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/8258
Part of https://github.com/vector-im/riot-web/issues/7606

image

### CODE REVIEW

**Comment by turt2live** [comment]
Once again, test failures are expected because of branch mismatches

File: src/components/views/settings/tabs/AdvancedRoomSettingsTab.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    constructor() {
+        super();
+
+        this.state = {
+            // This is eventually set to the value of room.getRecommendedVersion()
+            upgradeRecommendation: null,
+        };
+    }
+
+    componentWillMount(): void {
```
[the remark this answers is not in the document]
Comment (turt2live) [reply src/components/views/settings/tabs/AdvancedRoomSettingsTab.js]: Something my IDE autocompletes and I constantly forget to remove. I can remove it if wanted, and i should probably pay attention to what I'm overloading.

[the remark this answers is not in the document]
Comment (jryans) [reply src/components/views/settings/tabs/AdvancedRoomSettingsTab.js]: It seems a bit random to type only one method, so I guess I'd remove it... hopefully it's configurable in your IDE?

[the remark this answers is not in the document]
Comment (turt2live) [reply src/components/views/settings/tabs/AdvancedRoomSettingsTab.js]: probably. I write a lot of typescript and that's probably interfering with the jsx settings a bit. I'll just pay a bit more attention to it

**Pull request merged**